packages/core/src/services/documents/renameDocumentPaths.ts

```diff
@@ -0,0 +1,51 @@
+import type { Commit, DocumentVersion } from '../../browser'
+import { database } from '../../client'
+import { findWorkspaceFromCommit } from '../../data-access'
+import { Result, Transaction, TypedResult } from '../../lib'
+import { BadRequestError } from '../../lib/errors'
+import { DocumentVersionsRepository } from '../../repositories'
+import { updateDocument } from './update'
+
+export async function renameDocumentPaths(
+  {
+    commit,
+    oldPath,
+    newPath,
+  }: {
+    commit: Commit
+    oldPath: string
+    newPath: string
+  },
+  db = database,
+): Promise<TypedResult<DocumentVersion[], Error>> {
+  return await Transaction.call(async (tx) => {
+    if (commit.mergedAt !== null) {
+      return Result.error(new BadRequestError('Cannot modify a merged commit'))
+    }
+
+    const workspace = await findWorkspaceFromCommit(commit, tx)
```

Comment:
Pass workspace, don't use this

Reply:
this is valid actually, no need to pass workspace 🤷🏼 the equivalent of doing `commit.workspace` in rails

Reply:
uff ya. ok

Reply:
more performance to just pass it, anyways. right?

Reply:
yup but it results in an API that has invalid cases (I could pass one workspace and a commit from another workspace) and is non-obvious (if I'm already passing a commit from a workspace, why do i also have to pass the workspace?). The better solution is to fetch the tenant information together with the commit in the commits repository, so that getting the tenant from a commit does not result in an extra roundtrip to the db. later on. 

In any case we are splitting hairs here.